helix-term/src/ui/completion.rs

```diff
@@ -17,6 +17,9 @@ use crate::ui::{menu, Markdown, Menu, Popup, PromptEvent};
 
 use helix_lsp::{lsp, util, OffsetEncoding};
 
+const WEIGHT_SCORE_STEP: u32 = 1_000_000;
```

Comment:
Messing with the scoring of a fuzzy marcher is never a good idea. Consider them implementation details. The sorting by server should only be done as a tie breaker (like you did previously but instead or raking by server first and then by score it should be the other way around)